packages/components/src/Modal/GenericModal/GenericModal.tsx

```diff
@@ -14,6 +14,8 @@ export type GenericModalProps = {
   onEscapeKeyup?: (event: KeyboardEvent) => void
   onOutsideModalClick?: (event: React.MouseEvent) => void
   onAfterLeave?: () => void
+  /** Override default focus to the modal title and shift focus to this element after the modal is opened */
+  onOpenFocusTo?: HTMLElement | null
```

Comment:
wouldn't be better to pass just `onOpen` callback and allow the consumer to implement any behaviour they want, including focus? And we can have a story example how to use that for focus.

Reply:
Would much prefer to have a sensible default for accessibility with an escape hatch for the odd case. 9 times out of 10 we want focus to go to the heading, but most engineers & designers don't know that and will think it should just go to the first input because they don't understand the screen reader implications.

Reply:
totally we should have sensible default. Would having generic `onOpen` interface prevent that?

Reply:
I've updated the description of the PR but to quickly cover it here / hopefully allay concerns, I've gone with providing a callback that a consumer can user. If its called there's now a check that run in the `focusOnAccessibleLabel` function to see if focus has been shifted within the modal layer, if not shift focus to the accessible title as default. Going for a best of both worlds scenario here :)